Update the name of the breakpoint mixin to show that its deprecated

scruffian opened the pull request:
#### Changes proposed in this Pull Request

* Based on p4TIVU-9pb-p2 let's rename this mixin to make it clear that it's deprecated
* I looked into adding a stylelint rule, but it looked complicated so I took the easy route and just renamed the mixin :)


The reason behind this:
- We have virtually the same breakpoints in Calypso and Gutenberg but using different code.
- By using the same breakpoints we'll keep our CSS smaller and have a more consistent experience.
- We also keep our code simpler.

#### Testing instructions

* Look for uses of this mixin that I missed (I did a find and replace on `include breakpoint`);
* Test some pages to make sure they are still responsive.
* Check that the docs change is helpful

scruffian committed:
Update the name of the breakpoint mixin to show that its deprecated

simison commented:
Love it! <3

gwwar commented:
Did we get a design 👍 from the thread? I would also recommend updating the summary of this PR for reasons on this change and adding a custom stylelint plugin over a rename. Most folks don't read this sort of thing, and will copy paste this into other areas.

rickybanister commented:
I don't think it should hold up this PR, but there are a few areas where we'll want to tweak breakpoints to do things like hide illustrations etc:

![image](https://user-images.githubusercontent.com/437258/81434054-89d4b680-9133-11ea-905c-0c84698d224b.png)

And the single column layout highlights other design inconsistencies (but these already existed):

![image](https://user-images.githubusercontent.com/437258/81434228-d7e9ba00-9133-11ea-95c3-d7eb029a1426.png)

Thanks for doing this @scruffian!

cc @danhauk

jancavan commented:
> I don't think it should hold up this PR, but there are a few areas where we'll want to tweak breakpoints to do things like hide illustrations etc:

@rickybanister Did you scale your browser? I saw that happen too, but it's because we use JS viewports instead of CSS brekapoints. See: https://github.com/Automattic/wp-calypso/pull/41781#discussion_r419839462

scruffian commented:
This PR shouldn't change any functionality, so if you're seeing something different in this branch to in production then something is wrong :)

rickybanister commented:
Understood, thanks @jancavan 

@scruffian yes, I understood that. I didn't notice anything different, I simply noticed a few things I hadn't seen before in production or elsewhere.

Copons reviewed (dismissed):
This LGTM, I couldn't find any other use of the `breakpoint` mixin that you might have missed.

Nitpick: you did miss it in two comments in the `_breakpoints.scss` file:
https://github.com/Automattic/wp-calypso/blob/061b92b40616ed85b34cf887f1802d687cb86b7b/client/assets/stylesheets/shared/mixins/_breakpoints.scss#L47
https://github.com/Automattic/wp-calypso/blob/061b92b40616ed85b34cf887f1802d687cb86b7b/client/assets/stylesheets/shared/mixins/_breakpoints.scss#L55
I'd update them as well to avoid confusion.

scruffian committed:
updating the docs

saramarcondes commented on docs/coding-guidelines/css.md:
````diff
@@ -267,7 +224,7 @@ selector {
 - DO use a single space before an opening brace
 
 ```scss
-@include breakpoint( ">480px" ) {
````
Nit: could we change this example to not use a deprecated function?

saramarcondes commented on docs/coding-guidelines/css.md:
```diff
@@ -98,52 +98,9 @@ Under the hood, we are using webpack and its `sass-loader`, for compiling the st
 
 ## Media Queries
 
-We don't do device specific breakpoints, we use layout-specific breakpoints that communicate the state of the UI. DO NOT define your own media queries. Utilize the mixins [provided in `_mixins.scss`](https://github.com/Automattic/wp-calypso/blob/master/assets/stylesheets/shared/mixins/_breakpoints.scss). Furthermore, we are pushing for a mobile-first approach to media queries, meaning your defaul ...
```
I think this some of this documentation might actually still be useful, especially for devs who don't have extensive browser-client experience. Do you think we could just translate the uses of `breakpoint` to `break-*` instead of removing it completely? Or else, what's the reasoning for removing it entirely?

As an aside, it's nice that we're removing some room for variations in how breakpoints are applied given that the GB breakpoints are more restrictive (doesn't look like there's a `<` vs `>` problem); however, I think the approach documented in the bad/good section below is still valid.

saramarcondes commented on docs/coding-guidelines/css.md:
```diff
@@ -98,52 +98,9 @@ Under the hood, we are using webpack and its `sass-loader`, for compiling the st
 ## Media Queries
 
-We don't do device specific breakpoints, we use layout-specific breakpoints that communicate the state of the UI. DO NOT define your own media queries. Utilize the mixins [provided in `_mixins.scss`](https://github.com/Automattic/wp-calypso/blob/master/assets/stylesheets/shared/mixins/_breakpoints.scss). Furthermore, we are pushing for a mobile-first approach to media queries, meaning your defaul ...
+To avoid code bloat and have a more consistent experience we should try to use the same breakpoints in all SCSS files.
```
Nit: we could be more assertive in this sentence if we wanted 😛 

```suggestion
To avoid code bloat and have a more consistent experience we use the same breakpoints in all SCSS files whenever possible.
```

saramarcondes commented on docs/coding-guidelines/css.md:
```diff
@@ -98,52 +98,9 @@ Under the hood, we are using webpack and its `sass-loader`, for compiling the st
-The value passed to this mixin is actually a string rather than a pixel value. Accepted values are: `"<X"`, `">X"`, or `"X-Y"` — where `X` and `Y` are valid breakpoints, for example `480px`. If you provide any other value to the mixin it will fail and give you a warning in the output from `yarn start`.
-
-Adding additional breakpoints should not be undertaken lightly.
+The old `breakpoint` mixin should be replaced with the `break-*` mixins from [Gutenberg](https://github.com/WordPress/gutenberg/blob/0f1f5e75408705f0ec014f5d2ea3d9fcc8a97817/packages/base-styles/_mixins.scss)
```
```suggestion
The old `breakpoint-deprecated` mixin should be replaced with the `break-*` mixins from [Gutenberg](https://github.com/WordPress/gutenberg/blob/0f1f5e75408705f0ec014f5d2ea3d9fcc8a97817/packages/base-styles/_mixins.scss)
```

sarayourfriend reviewed (commented):
Doc changes LGTM, just a couple suggestions and questions for clarification.

scruffian committed:
Update docs/coding-guidelines/css.md

Co-authored-by: Sara Marcondes <saram@fastmail.com>

scruffian committed:
Update docs/coding-guidelines/css.md

Co-authored-by: Sara Marcondes <saram@fastmail.com>

scruffian committed:
replace more uses of the deprecated breakpoint mixin

scruffian committed:
update to the docs

scruffian commented:
@saramarcondes Thanks for the review. I updated the docs and did a rebase, and then changed the new uses of the breakpoint mixin to be -deprecated.

I removed the part of the docs that talk about performance because we discovered that this isn't true (p4TIVU-9fo-p2/#comment-9880).

It would be great if we would get the merged ASAP as keeping it up to date is hard!
